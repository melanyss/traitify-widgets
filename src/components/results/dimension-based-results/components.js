export DimensionBasedResults from "./index";
export Dimension from "./dimension";
export Dimensions from "./dimensions";
export PersonalityDetails from "./personality-details";
export PersonalityHeading from "./personality-heading";
export PersonalityTrait from "./personality-trait";
export PersonalityTraits from "./personality-traits";
export PersonalityType from "./personality-type";
export PersonalityTypes from "./personality-types";
export Radar from "./radar";
