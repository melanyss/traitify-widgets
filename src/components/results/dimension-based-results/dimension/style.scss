@import "~style/helpers";

.dimension {
  font: 100%/1.5 "Source Sans Pro", "Helvetica Neue", Verdana, Arial, sans-serif;
  font-weight: 400;
  color: #444;
  max-width: $breakPoint5;
  margin: 0 auto;
  text-align: left;
  color: white;
  margin-bottom: 20px;
  &:last-of-type {
    margin-bottom: 0;
  }
  .trigger {
    color: white;
    text-decoration: none;
    display: inline-block;
    float: right;
    padding: 10px 20px 5px;
    i {
      font-size: 26px;
      vertical-align: middle;
    }
  }
  .main {
    &:after {
      clear: both;
      content: " ";
      display: table;
    }
    .content {
      color: #333;
      padding: 20px 20px 30px;
      @include mq(m) {
        margin-left: 170px;
        padding-top: 40px;
        text-align: left;
      }
      @include mq(l) {
        padding-right: 80px;
      }
      @include mq(xl) {
        padding-right: 170px;
      }
      h4 {
        margin: 0;
        font-weight: normal;
        text-align: center;
        @include rems(24);
        @include mq(l) {
          @include rems(28);
        }
        @include mq(xl) {
          @include rems(34);
        }
      }
      p {
        @include mq(m) {
          @include rems(17);
        }
        @include mq(l) {
          @include rems(18);
        }
        @include mq(xl) {
          @include rems(19);
        }
      }
    }
  }
  .side {
    clear: both;
    text-align: center;
    @include mq(m) {
      float: left;
      width: 130px;
      margin: 5px 20px 0;
    }
    .icon {
      @include mq(m) {
        margin-bottom: 10px;
      }
    }
    .icon img {
      width: 50%;
      max-width: 220px;
      @include mq(m) {
        width: 100%;
      }
    }
    .score {
      font-weight: bold;
      margin-top: 10px;
      @include rems(26);
      @include mq(l) {
        @include rems(28);
      }
    }
  }
  .details {
    clear: both;
    margin: 0 -20px;
    @include mq(xl) {
      margin: 0 60px;
    }
    @include mq(xxxl) {
      margin: 0 130px;
    }
    .content {
      padding: 20px;
      @include mq(m) {
        padding-left: 40px;
      }
      @include mq(xl) {
        padding-left: 0;
        border-bottom-left-radius: 25px;
        border-bottom-right-radius: 25px;
      }
    }
  }
  .detail {
    padding: 0 20px;
    @include mq(m) {
      display: inline-block;
      vertical-align: top;
      width: 50%;
      @include rems(17);
    }
    @include mq(xl) {
      width: 38%;
      &:first-of-type {
        margin-left: calc(28%/2);
      }
    }
    @include mq(xxxl) {
      width: 35%;
      &:first-of-type {
        margin-left: calc(33%/2);
      }
    }
    h5 {
      margin: 0;
      line-height: 35px;
      font-weight: normal;
      @include rems(18);
      @include mq(xl) {
        @include rems(20);
      }
      &:before {
        content: " ";
        display: inline-block;
        width: 35px;
        height: 35px;
        vertical-align: bottom;
        margin-right: 10px;
      }
    }
    ul {
      list-style: none;
      padding: 20px 0;
      margin: 0;
      @include mq(m) {
        padding: 10px 0;
        margin-left: 45px;
      }
    }
    li {
      margin-top: 5px;
    }
    .benefits:before {
      background: url("https://cdn.traitify.com/assets/images/thumbs_up.svg") no-repeat left center;
    }
    .pitfalls:before {
      background: url("https://cdn.traitify.com/assets/images/thumbs_down.svg") no-repeat left center;
    }
  }
}
