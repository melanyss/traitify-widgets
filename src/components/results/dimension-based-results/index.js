import {Component} from "react";
import Radar from "./radar";
import PersonalityTypes from "./personality-types";
import PersonalityHeading from "./personality-heading";
import PersonalityDetails from "./personality-details";
import Dimensions from "./dimensions";
import PersonalityTraits from "./personality-traits";

export default class DimensionBasedResults extends Component {
  render() {
    return (
      <section>
        <Radar {...this.props} />
        <PersonalityTypes {...this.props} />
        <PersonalityHeading {...this.props} />
        <PersonalityDetails {...this.props} />
        <Dimensions {...this.props} />
        <PersonalityTraits {...this.props} />
      </section>
    );
  }
}
