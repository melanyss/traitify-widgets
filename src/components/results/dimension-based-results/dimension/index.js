import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropType from "lib/helpers/prop-type";
import withTraitify from "lib/with-traitify";
import {rgba} from "lib/helpers/color";
import style from "./style";

class Dimension extends Component{
  static propTypes = {
    getOption: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    traitify: TraitifyPropType.isRequired,
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.object.isRequired,
      score: PropTypes.number.isRequired
    }).isRequired
  }
  constructor(props){
    super(props);

    this.state = {showContent: props.index === 0};
  }
  trigger = (e)=>{
    e.preventDefault();

    this.props.traitify.ui.trigger("Dimension.showContent", this, this.props.type.personality_type);
    this.setState((state)=>({showContent: !state.showContent}));
  }
  componentDidMount(){
    this.props.traitify.ui.trigger("Dimension.initialized", this);
  }
  componentDidUpdate(){
    this.props.traitify.ui.trigger("Dimension.updated", this);
  }
  description(suffix){
    const type = this.props.type.personality_type;
    const perspective = (this.props.getOption("perspective") || "firstPerson").replace("Person", "");
    let description = type.details.find((detail)=>(detail.title === `${perspective}_person_${suffix}`));
    description = description && description.body;

    if(description){ return description; }
    description = type.details.find((detail)=>(detail.title === `${perspective}_person_${suffix}`));

    return (description && description.body) || type.description;
  }
  render(){
    const type = this.props.type.personality_type;
    const color = `#${type.badge.color_1}`;
    const benefits = [];
    const pitfalls = [];

    type.details.forEach((detail)=>{
      if(detail.title === "Benefits"){ benefits.push(detail.body); }
      if(detail.title === "Pitfalls"){ pitfalls.push(detail.body); }
    });

    return (
      <li className={style.dimension}>
        <div className={style.main} style={{background: rgba(color, 8.5), borderLeft: `5px solid ${color}`}}>
          <div className={style.side}>
            <p className={style.icon}>
              <img src={type.badge.image_medium} alt="" ariahidden="true" />
            </p>
          </div>
          <div className={style.content}>
            <h2 className={style.title}>
              {type.name} <span style={{color}}>|</span> {this.props.type.score} - {type.level}
            </h2>
            <p className={style.description}>{this.description("short_description")}</p>
            <p className={style.triggerButton}>
              <button
                className={style.trigger}
                style={{background: rgba(color, 30)}}
                onClick={this.trigger}
                type="button"
              >
                {this.props.translate(this.state.showContent ? "show_less" : "show_more")}
              </button>
            </p>
          </div>
        </div>
        {this.state.showContent && (
          <div className={style.details}>
            <div className={style.content} style={{background: rgba(color, 30)}}>
              <div className={style.extendedDesc}>
                <h3>{this.props.translate("extended_description")}</h3>
                <p className={style.description}>{this.description("description")}</p>
              </div>
              <div className={style.detail}>
                <h4 className={style.benefits}>{this.props.translate("potential_benefits")}</h4>
                <ul>
                  {benefits.map((benefit)=>(<li key={benefit}>{benefit}</li>))}
                </ul>
              </div>
              <div className={style.detail}>
                <h4 className={style.pitfalls}>{this.props.translate("potential_pitfalls")}</h4>
                <ul>
                  {pitfalls.map((pitfall)=>(<li key={pitfall}>{pitfall}</li>))}
                </ul>
              </div>
            </div>
          </div>
        )}
      </li>
    );
  }
}

export {Dimension as Component};
export default withTraitify(Dimension);
