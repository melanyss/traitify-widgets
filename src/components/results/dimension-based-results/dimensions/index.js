import Component from "components/traitify-component";
import Dimension from "../dimension";
import style from "./style";

export default class Dimensions extends Component{
  componentDidMount(){
    this.traitify.ui.trigger("Dimensions.initialized", this);
    this.followAssessment();
  }
  componentDidUpdate(){
    this.followAssessment();
  }
  render(){
    if(!this.isReady("results")){ return; }

    const options = this.copyOptions();

    return (
      <ul class={style.dimensions}>
        {this.state.assessment.personality_types.map((type, i)=>{
          return <Dimension type={type} index={i} options={options} />;
        })}
      </ul>
    );
  }
}
