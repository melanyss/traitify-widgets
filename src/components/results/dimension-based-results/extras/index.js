import {h, Component} from "preact";
import style from "./style";

export default class Extras extends Component{
  componentDidMount(){
    this.props.triggerCallback("Extras", "initialized", this);
  }
  render(){
    if(!this.props.resultsReady()) return <div />;

    let props = this.props;
    return (
			<div class={style.extras}>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #008dc7;`}></div>
						<h4 class={style.title} style={`color: #008dc7;`}>Complements</h4>
						<p class={style.description}>Sit harum culpa repellendus natus placeat cumque Non dignissimos fugiat sit quos pariatur. Rem distinctio laudantium illo fugiat maxime, animi. Laudantium quasi ex iure recusandae animi. Perferendis nemo dolores ipsam. Dolor optio accusantium quis modi pariatur! Accusantium ad cum repellat sit corrupti in consequuntur Consequatur quia vitae deserunt quos esse. Tempore consequatur id veniam eaque dolor odit Nemo dolores aperiam?</p>
					</div>
				</div>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #d04e4a;`}></div>
						<h4 class={style.title} style={`color: #d04e4a;`}>Conflicts</h4>
						<p class={style.description}>Sit harum culpa repellendus natus placeat cumque Non dignissimos fugiat sit quos pariatur. Rem distinctio laudantium illo fugiat maxime, animi. Laudantium quasi ex iure recusandae animi. Perferendis nemo dolores ipsam. Dolor optio accusantium quis modi pariatur! Accusantium ad cum repellat sit corrupti in consequuntur Consequatur quia vitae deserunt quos esse. Tempore consequatur id veniam eaque dolor odit Nemo dolores aperiam?</p>
					</div>
				</div>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #32be4b;`}></div>
						<h4 class={style.title} style={`color: #32be4b;`}>Best Work Environments</h4>
						<ul class={style.description}>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
						</ul>
					</div>
				</div>
			</div>
    );
  }
}
