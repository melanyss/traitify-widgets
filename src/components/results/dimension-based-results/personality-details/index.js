import {Component} from "preact";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component{
  componentDidMount(){
    this.props.traitify.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate(){
    this.props.traitify.ui.trigger("PersonalityDetails.updated", this);
  }
  render(){
    if(!this.props.isReady("results")){ return; }

    const personality = this.props.assessment.archetype || {};
    const details = personality.details;
    if(!details){ return; }

    let complement = details.find(d=>d.title === "Complement");
    complement = complement && complement.body;
    let conflict = details.find(d=>d.title === "Conflict");
    conflict = conflict && conflict.body;
    const environments = personality.environments || [];

    return (
      <div class={style.details}>
        {complement && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #008dc7;" />
              <h4 class={style.title} style="color: #008dc7;">{this.props.translate("complements")}</h4>
              <p class={style.description}>{complement}</p>
            </div>
          </div>
        )}
        {conflict && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #d04e4a;" />
              <h4 class={style.title} style="color: #d04e4a;">{this.props.translate("conflicts")}</h4>
              <p class={style.description}>{conflict}</p>
            </div>
          </div>
        )}
        {environments[0] && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #32be4b;" />
              <h4 class={style.title} style="color: #32be4b;">{this.props.translate("best_work_environments")}</h4>
              <ul class={style.description}>
                {environments.map(environment=>(
                  <li>{environment.name}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
