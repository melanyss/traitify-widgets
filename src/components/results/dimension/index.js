import { h, Component } from "preact";
import style from "./style";

export default class Dimension extends Component {
  constructor(props) {
    super(props);
    this.state = {showContent: props.index == 0}
  }
  trigger(context, e) {
    e.preventDefault();
    context.setState({showContent: !context.state.showContent});
  }
	convertHex(hex,opacity) {
		hex = hex.replace('#','');
		var r = parseInt(hex.substring(0,2), 16);
		var g = parseInt(hex.substring(2,4), 16);
		var b = parseInt(hex.substring(4,6), 16);
		var result = 'rgba('+r+','+g+','+b+','+opacity/100+')';
		return result;
	}
  render() {
    var type = this.props.personalityType.personality_type;
    var benefits = [];
    var pitfalls = [];
    type.details.forEach(function(detail) {
      if(detail.title == "Benefits") benefits.push(detail.body)
      if(detail.title == "Pitfalls") pitfalls.push(detail.body)
    });
    var color = `#${type.badge.color_1}`;
    var color2 = `#${type.badge.color_2}`;
    return (
      <li class={style.dimension} style={`border-top: 3px solid ${color};`}>
        <div class={style.main} style={`background: ${this.convertHex(color,8.5)};`}>
          <a class={style.trigger} style={`color: ${color};`} onClick={this.trigger.bind(null, this)} href="#">{this.props.translate(this.state.showContent ? "show_less" : "show_more")}</a>
          <div class={style.side}>
            <p class={style.icon}>
              <img src={type.badge.image_medium} alt={type.name} />
            </p>
            <p class={style.score}>{this.props.personalityType.score} - {type.level}</p>
          </div>
          <div class={style.content}>
            <h4 class={style.title}>{type.name}</h4>
            <p class={style.description}>{type.description}</p>
						<p class={style.center}><a class={style.lowerTrigger} style={`background: ${this.convertHex(color,30)}`} onClick={this.trigger.bind(null, this)} href="#">{this.props.translate(this.state.showContent ? "show_less" : "show_more")}</a></p>
          </div>
        </div>
        {this.state.showContent &&
          <div class={style.details}>
            <div class={style.content} style={`background: ${this.convertHex(color,30)}`}>
              <div class={style.detail}>
                <h5 class={style.benefits}>{this.props.translate("potential_benefits")}</h5>
                <ul>
                  {benefits.map(function(benefit) {
                    return <li>{benefit}</li>
                  })}
                </ul>
              </div>
              <div class={style.detail}>
                <h5 class={style.pitfalls}>{this.props.translate("potential_pitfalls")}</h5>
                <ul>
                  {pitfalls.map(function(pitfall) {
                    return <li>{pitfall}</li>
                  })}
                </ul>
              </div>
            </div>
          </div>
        }
      </li>
    );
  }
}
