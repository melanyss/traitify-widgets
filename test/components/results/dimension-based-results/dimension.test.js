import {Component} from "components/results/dimension-based-results/dimension";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("Dimension", () => {
  let props;

  beforeEach(() => {
    props = {
      getOption: jest.fn().mockName("getOption"),
      index: 1,
      translate: jest.fn().mockName("translate"),
      type: assessment.personality_types[0],
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger.mock.calls[0][0]).toBe("Dimension.initialized");
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger.mock.calls[1][0]).toBe("Dimension.updated");
    });

    it("triggers show content", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.trigger();

      expect(props.ui.trigger.mock.calls[1][0]).toBe("Dimension.showContent");
      expect(props.ui.trigger.mock.calls[1][2]).toBe(props.type.personality_type);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  describe("with index of 0", () => {
    beforeEach(() => {
      props.index = 0;
    });

    it("shows content by default", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(component.state.showContent).toBe(true);
      expect(component.tree).toMatchSnapshot();
    });

    it("trigger toggles content", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.trigger();

      expect(component.state.showContent).toBe(false);
    });

    it("assessment change resets content", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.trigger();
      component.updateProps({assessmentID: "other"});

      expect(component.state.showContent).toBe(true);
    });
  });

  describe("with non-zero index", () => {
    it("hides content by default", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(component.state.showContent).toBe(false);
      expect(component.tree).toMatchSnapshot();
    });

    it("trigger toggles content", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.trigger();

      expect(component.state.showContent).toBe(true);
    });

    it("assessment change resets content", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.instance.trigger();
      component.updateProps({assessmentID: "other"});

      expect(component.state.showContent).toBe(false);
    });
  });

  describe("description", () => {
    it("defaults to first person", () => {
      const component = new ComponentHandler(<Component {...props} />);
      const description = props.type.personality_type.details.find((detail) => (detail.title === "first_person_description"));

      expect(component.instance.description("description")).toBe(description.body);
    });

    it("shows third person", () => {
      props.getOption.mockReturnValue("thirdPerson");
      const component = new ComponentHandler(<Component {...props} />);
      const description = props.type.personality_type.details.find((detail) => (detail.title === "third_person_description"));

      expect(component.instance.description("description")).toBe(description.body);
    });

    it("falls back to third person", () => {
      const details = props.type.personality_type.details.filter((detail) => (detail.title !== "first_person_description"));
      props.type = {
        ...props.type,
        personality_type: {
          ...props.type.personality_type,
          details
        }
      };
      const component = new ComponentHandler(<Component {...props} />);
      const description = props.type.personality_type.details.find((detail) => (detail.title === "third_person_description"));

      expect(component.instance.description("description")).toBe(description.body);
    });

    it("falls back to first person", () => {
      props.getOption.mockReturnValue("thirdPerson");
      const details = props.type.personality_type.details.filter((detail) => (detail.title !== "third_person_description"));
      props.type = {
        ...props.type,
        personality_type: {
          ...props.type.personality_type,
          details
        }
      };
      const component = new ComponentHandler(<Component {...props} />);
      const description = props.type.personality_type.details.find((detail) => (detail.title === "first_person_description"));

      expect(component.instance.description("description")).toBe(description.body);
    });

    it("falls back to description", () => {
      const details = props.type.personality_type.details
        .filter((detail) => (detail.title !== "first_person_description"))
        .filter((detail) => (detail.title !== "third_person_description"));
      props.type = {
        ...props.type,
        personality_type: {
          ...props.type.personality_type,
          details
        }
      };
      const component = new ComponentHandler(<Component {...props} />);
      const {description} = props.type.personality_type;

      expect(component.instance.description("description")).toBe(description);
    });
  });
});
