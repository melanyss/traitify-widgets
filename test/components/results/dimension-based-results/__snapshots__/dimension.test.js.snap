// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dimension renders component 1`] = `
<li>
  <div
    style={
      Object {
        "background": "rgba(41,183,112,0.085)",
        "borderLeft": "5px solid #29B770",
      }
    }
  >
    <div>
      <p>
        <img
          alt=""
          ariahidden="true"
          src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
        />
      </p>
    </div>
    <div>
      <h2>
        Agreeableness
         
        <span
          style={
            Object {
              "color": "#29B770",
            }
          }
        >
          |
        </span>
         
        6
         - 
        Medium
      </h2>
      <p>
        Influencing people (Agreeableness) concerns the way someone balances their emotional understanding of other people, and their respect for differing viewpoints, with the style in which they try to influence or negotiate with them.
      </p>
      <p>
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(41,183,112,0.3)",
            }
          }
          type="button"
        />
      </p>
    </div>
  </div>
</li>
`;

exports[`Dimension with index of 0 shows content by default 1`] = `
<li>
  <div
    style={
      Object {
        "background": "rgba(41,183,112,0.085)",
        "borderLeft": "5px solid #29B770",
      }
    }
  >
    <div>
      <p>
        <img
          alt=""
          ariahidden="true"
          src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
        />
      </p>
    </div>
    <div>
      <h2>
        Agreeableness
         
        <span
          style={
            Object {
              "color": "#29B770",
            }
          }
        >
          |
        </span>
         
        6
         - 
        Medium
      </h2>
      <p>
        Influencing people (Agreeableness) concerns the way someone balances their emotional understanding of other people, and their respect for differing viewpoints, with the style in which they try to influence or negotiate with them.
      </p>
      <p>
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(41,183,112,0.3)",
            }
          }
          type="button"
        />
      </p>
    </div>
  </div>
  <div>
    <div
      style={
        Object {
          "background": "rgba(41,183,112,0.3)",
        }
      }
    >
      <div>
        <h3 />
        <p>
          Influencing people (Agreeableness) concerns the way someone balances their emotional understanding of other people, and their respect for differing viewpoints, with the style in which they try to influence or negotiate with them.
        </p>
      </div>
      <div>
        <h4 />
        <ul>
          <li>
            Forgiving style
          </li>
          <li>
            Looks for win-win outcome
          </li>
          <li>
            Negotiates with sensitivity
          </li>
          <li>
            Realistic view of others
          </li>
          <li>
            Recognises imperfection 
          </li>
          <li>
            Relational style
          </li>
        </ul>
      </div>
      <div>
        <h4 />
        <ul>
          <li>
            May appear too open-minded (to others)
          </li>
          <li>
            May be viewed as a ‘pushover’ by others’
          </li>
          <li>
            May let sympathy get in the way
          </li>
          <li>
            May lose out to very assertive colleagues
          </li>
          <li>
            May not always be objective enough
          </li>
          <li>
            May not go for the optimal personal win
          </li>
        </ul>
      </div>
    </div>
  </div>
</li>
`;

exports[`Dimension with non-zero index hides content by default 1`] = `
<li>
  <div
    style={
      Object {
        "background": "rgba(41,183,112,0.085)",
        "borderLeft": "5px solid #29B770",
      }
    }
  >
    <div>
      <p>
        <img
          alt=""
          ariahidden="true"
          src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
        />
      </p>
    </div>
    <div>
      <h2>
        Agreeableness
         
        <span
          style={
            Object {
              "color": "#29B770",
            }
          }
        >
          |
        </span>
         
        6
         - 
        Medium
      </h2>
      <p>
        Influencing people (Agreeableness) concerns the way someone balances their emotional understanding of other people, and their respect for differing viewpoints, with the style in which they try to influence or negotiate with them.
      </p>
      <p>
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(41,183,112,0.3)",
            }
          }
          type="button"
        />
      </p>
    </div>
  </div>
</li>
`;
